File: BOJ/Graph/Silver/1697/1697.py
from collections import deque
n,k = map(int,input().split())
visit = [0 for i in range(100001)]
q = deque()
q.append((n,0))
while len(q) > 0 :
    curx, cnt = q.popleft()
    if curx == k :
        print(cnt)
        break
    nx = [curx+1,curx-1,curx*2]
    for next in nx :
        if next < 0 or next > 100000 : continue
        if visit[next] != 0 : continue
        visit[next] = cnt+1
        q.append((next,cnt+1))

File: BOJ/Graph/Silver/1697/1697.swift
let input = readLine()!.split(separator: " ").map{Int(String($0))!}
let (n,k) = (input[0],input[1])
var visit = Array(repeating : 0, count : 100001)
var q = [(Int,Int)]()
q.append((n,0))
while !q.isEmpty {
    let (curx, cnt) = q.removeFirst()
    if curx == k {
        print(cnt)
        break
    }
    let nx = [curx+1,curx-1,curx*2]
    for next in nx {
        if next < 0 || next > 100000 {continue}
        if visit[next] != 0 {continue}
        visit[next] = cnt+1
        q.append((next,cnt+1))
    }
}
